CRM/MembershipExtras/Hook/Pre/MembershipPaymentPlanProcessor.php

```diff
@@ -114,10 +114,24 @@ private function alterFirstContributionParameters() {
     $this->params['net_amount'] =  $this->recurringContribution['amount'];
 
     if (!empty($this->params['tax_amount'])) {
-      $this->params['tax_amount'] = $this->calculateSingleInstallmentAmount($this->params['tax_amount']);
+      $this->params['tax_amount'] = $this->calculateInstallmentTax($this->params['total_amount']);
```

Comment:
this code works for both price-set and financial type membership payments, doing this will screw priceset payments, so you need to do this only if it this not a price-set payment.

Reply:
Well I tested it with a Priceset and it all seemed ok, but I'll look into it a little deeper just to be safe.

Reply:
Yeah, it all works correctly. Basically it just recalculates tax amount from the actual value and tax rate, which I think is a good thing, both for the contribution and for each line item in the price set.

![image](https://user-images.githubusercontent.com/21999940/42331741-d24ac07c-803b-11e8-9a48-1f10f78f6114.png)

Reply:
it is a problematic  because of this : 

```
    $taxRates = CRM_Core_PseudoConstant::getTaxRates();
    $rate = CRM_Utils_Array::value($this->params['financial_type_id'], $taxRates, 0);
```

you basically use the contribution financial_type_id which is something does not exist when paying using price sets, assume you have 3 line items (price fields) in the price set where each one has a different financial type (and each financial type has a different amount of tax),  but code above will use single tax rate for all the line items (price fields) even if they have different amount of taxes, or at least this is what I can tell from reading the code.